spec/models/article_spec.rb

```diff
@@ -613,5 +613,33 @@ def build_and_validate_article(*args)
     end
   end
 
+  describe "#active_published_articles" do
+    let(:not_archived_not_published_article) { create(:article, archived: false, published: false) }
+    let(:archived_not_published_article) { create(:article, archived: true, published: false) }
+    let(:not_archived_published_article) { create(:article, archived: false, published: true) }
+    let(:archived_published_article) { create(:article, archived: true, published: true) }
+
+    before do
+      not_archived_not_published_article.validate
+      archived_not_published_article.validate
+      not_archived_published_article.validate
+      archived_published_article.validate
+    end
```

Comment:
Why do we need these? :)

Reply:
@rhymes It's my bad. Because I have used `let`, somehow I want to initialise it unnecessarily. Removed those by using `let_it_be`.